handbook/engineering/distribution/managed/upgrade_process.md

```diff
@@ -245,48 +258,190 @@ During this time users will see 500 errors:
+
+## Machine upgrade
+
+This is the upgrade process for changing the underlying machine for a managed instance.
+
+### 0) Machine upgrade setup
+
+**Follow the [general setup](#general-setup) guide.** Then set up a date to identify your upgrade:
 
 ```sh
-../util/set-notice.sh none
+export DATE=$(date +%m-%d-%Y)
 ```
 
-## 11) Take down the old deployment
+Validate all variables are set:
 
-Remove the old `$OLD_DEPLOYMENT` deployment and its data disk:
+```sh
+./util/validate-env.ts
+```
+
+Then set up a branch for your changes:
 
 ```sh
-../util/drop-deployment.ts $OLD_DEPLOYMENT drop-disk
-rm -rf $OLD_DEPLOYMENT/
-git add . && git commit -m "$CUSTOMER: remove $OLD_DEPLOYMENT deployment"
+git checkout -b $CUSTOMER/machine-upgrade-$DATE
+# all the below steps are documented assuming you are in the customer deployment directory
+cd $CUSTOMER
 ```
 
-## 13) Open a pull request to commit your changes
+### 1) Prepare for maintainence
+
+Set up access to the frontend by copying this output and running it in another shell:
 
 ```sh
-git push origin HEAD
+echo "gcloud compute start-iap-tunnel default-$OLD_DEPLOYMENT-instance 80 --local-host-port=localhost:4444 --zone us-central1-f --project sourcegraph-managed-$CUSTOMER"
 ```
 
-And click the provided link to open a pull request in [`deploy-sourcegraph-managed`](https://github.com/sourcegraph/deploy-sourcegraph-managed).
+Note that maintainence is being performed:
 
-**IMPORTANT: DO NOT FORGET TO GET YOUR PR APPROVED AND MERGED**, if you forget then the next person upgrading the instance will have a very bad time.
+```sh
+../util/set-notice.sh maintainence
+```
+
+Then, [mark the database as read-only](#2-mark-the-database-as-ready-only).
+
+Create a snapshot:
+
+```sh
+../util/create-snapshot.ts $OLD_DEPLOYMENT upgrade-machine-$DATE
+git add . && git commit -m "$CUSTOMER: snapshot deployment"
+```
+
+This can take anywhere from a minute to several minutes, depending on how large the disk is.
+
+### 2) Prepare resource changes in Terraform
+
+For example, in `terraform.tfvars`:
+
+```diff
+- machine_type         = "n1-standard-8"
++ machine_type         = "n1-standard-32"
+- data_disk_size       = 250 # GB
++ data_disk_size       = 500 # GB
+```
+
+Notes:
+
+- See [Cost estimation: working out the VM type required](cost_estimation.md#working-out-the-vm-type-required) for details about what you can choose for `machine_type`.
+- `data_disk_size` can only increase, it can never be decreased.
+
+### 3) Spin up an upgraded machine
+
+Set up configuration for the new machine:
+
+```sh
+cp -R $OLD_DEPLOYMENT/ $NEW_DEPLOYMENT/
+git add $NEW_DEPLOYMENT/ && git commit -m "$CUSTOMER: set up $NEW_DEPLOYMENT configuration"
+```
+
+Initialize the new production deployment (`NEW_DEPLOYMENT`) using the snapshot created in the previous step.
+**⚠️ When the tool prompts you to apply, exit the command (DO NOT APPLY THE CHANGES). ⚠️**
```

Comment:
I'm wondering, would it be possible to fix that with a flag, like making https://github.com/sourcegraph/deploy-sourcegraph-managed/blob/main/util/init-deployment.ts#L30 conditional to that flag? That'd make this step less scary 😅 !

Reply:
I think the terraform command stopping to ask you for a `yes` confirmation is pretty close to the effectiveness of such a flag. This is the only place it should happen, and machine upgrades happen rarely (and presumably by someone who's done a normal upgrade before so is familiar), so I think i'd rather avoid potentially missing the apply in the normal upgrade process - WDYT?